sky/clouds/gcp.py

```diff
@@ -141,8 +141,27 @@ def region_zones_provision_loop(
             assert len(accelerators) == 1, accelerators
             acc = list(accelerators.keys())[0]
             acc_count = list(accelerators.values())[0]
-            regions = service_catalog.get_region_zones_for_accelerators(
+            acc_regions = service_catalog.get_region_zones_for_accelerators(
                 acc, acc_count, use_spot, clouds='gcp')
+            if instance_type == 'TPU-VM':
+                regions = acc_regions
+            else:
+                # Host VM and accelerator must be in the same region and zone.
```

Comment:
Is there a test case that this change fixes?

Reply:
No. One consequence of this is that the TPU Nodes in Europe and Asia will not be supported, because our GCP catalog does not include host VMs (i.e., N1 machines) in the regions outside the US.

Reply:
@concretevitamin This is still an unsolved problem. To my understanding, TPUs in Europe and Asia are currently supported due to our bugs. If this PR is merged, those instances will be not supported. Should we add N1 machines in Europe and Asia to the catalog?